internal/testing/test.go

```diff
@@ -0,0 +1,85 @@
+package testing
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Joffref/genz/internal/utils"
+	"os"
+	"path"
+	"strings"
+)
+
+func RunTests(directory string, verbose, exitOnError bool) error {
+	entries, err := os.ReadDir(directory)
+	if err != nil {
+		return fmt.Errorf("failed to read directory %s: %w", directory, err)
+	}
+	var subDirectories []string
+	for _, f := range entries {
+		if f.IsDir() {
+			subDirectories = append(subDirectories, f.Name())
+			continue
+		}
+		if f.Name() == "expected.go" {
+			if err := runTest(directory, verbose); err != nil {
+				return err
+			}
+		}
+	}
+	var errs error
+	for _, name := range subDirectories {
+		if err := RunTests(path.Join(directory, name), verbose, exitOnError); err != nil {
+			if exitOnError {
+				return err
+			}
+			errs = errors.Join(errs, err)
+		}
+	}
+	return errs
+}
+
+func runTest(directory string, verbose bool) error {
+	fmt.Println("running test in", directory)
+	if err := utils.RunCommand([]string{"go", "generate", fmt.Sprintf("./%s", directory)}, verbose); err != nil {
+		return err
+	}
+	expected, err := os.ReadFile(path.Join(directory, "expected.go"))
+	if err != nil {
+		return fmt.Errorf("failed to read expected.go file: %w", err)
+	}
+	dir, err := os.ReadDir(directory)
+	if err != nil {
+		return err
+	}
+	if len(dir) == 0 {
+		return fmt.Errorf("no file in directory %s", directory)
+	}
+	var generatedFiles []string
+	for _, f := range dir {
+		if strings.HasSuffix(f.Name(), ".gen.go") {
+			generatedFiles = append(generatedFiles, f.Name())
```

Comment:
Store the path to the file, this makes the work after easier, avoiding to pass variables and recompute the path every time
```suggestion
			generatedFiles = append(generatedFiles, path.Join(directory, f.Name()))
```
Then, you can refactor
```go
func cleanUpGeneratedFiles(directory string, generatedFiles []string) error {
// to
func removeFiles(paths []string) error {
```